docs/public/components/using-components.md

```diff
@@ -0,0 +1,169 @@
+                )
+            },
+        }
+
+        return []*cobra.Command{cmd}
+    }
+    ```
+
+The code above creates a test command that does nothing. You can see we're using the `fxutil.OneShot` helpers. This
+helpers will initialize an `Fx` app for us with all the wanted dependencies. The next section explain how to request a
+dependency.
+
+## Importing components
+
+The `fxutil.OneShot` will take a list of components and give them to `Fx`. It's important to understanding that this
+will only tell `Fx` how to create types when they're needed. This alone will do nothing more.
+
+In order for a components to be instantiated it needs to be required:
+
++ As a parameter for the `callback` function
++ As dependencies from other components already marked for instantiation
++ Directly asked for by using `fx.Invoke`. More on this in [fx page](fx.md).
+
+Let's require the `log` components:
+
+```go
+import (
+    // First let's import the FX wrapper to require it
+    logfx "github.com/DataDog/datadog-agent/comp/core/log/fx"
+    // Then the logger interface to use it
+    log "github.com/DataDog/datadog-agent/comp/core/log/def"
+)
+
+// [...]
+    return fxutil.OneShot(
+        myTestCallback, // The function to call from fxutil.OneShot
+        logfx.Module(), // This will tell FX how to create the `log.Component`
+    )
+// [...]
+
+func myTestCallback(logger log.Component) {
+    logger.Info("some message")
+}
+```
+
+## Importing bundles
+
+Now let's say we want to include the core bundle instead. The core bundle offers many basic features (logger, config,
+telemetry, flare, ...).
+
+```go
+import (
+    // We import the core bundle
+    core "github.com/DataDog/datadog-agent/comp/core"
+
+    // Then the interfaces we want to use
+    config "github.com/DataDog/datadog-agent/comp/core/config/def"
+)
+
+// [...]
+    return fxutil.OneShot(
+        myTestCallback, // The function to call from fxutil.OneShot
+        core.Bundle(),  // This will tell FX how to create the all the components included in the bundle
+    )
+// [...]
+
+func myTestCallback(conf config.Component) {
+    api_key := conf.GetString("api_key")
+
+    // [...]
+}
+```
+
+It's very important to understand that since `myTestCallback` only uses the `config.Component` not all components from
+the `core` bundle was instantiated ! The `core.Bundle` instruct `Fx` how to create components but only the ones required
+are created.
+
+In our example, the `config.Component` might have dozens of dependencies instantiated from the core bundle. `Fx` handles
+all of this for us.
+
+## Using non-component types with Fx
+
+As our migration to components is not finished you might need to manually instruct `Fx` on how to create certain types.
+
+You will need to use `fx.Provide` for this. More details can be found [here](fx.md).
+
+But here is a quick example:
+
+```go
+import (
+    logfx "github.com/DataDog/datadog-agent/comp/core/log/fx"
```

Comment:
This makes me think: what if we require all `fx` packages to alias the interface type. So `fx/fx.go` would contain `type Component = def.Compoent`. That would keep type safety, while removing one import, and simplifying the alias usage in clients like this.

Reply:
def will also contain all the public types returned/used by the interface. I don't think we want to start down the aliasing path.